File: Qwirkle.AI/TileOnPlayer.py
from Tile import Tile
from Tile import Tile
from Shape import Shape
from Color import Color

class TileOnPlayer(Tile):
    def __init__(self, tile):
        self.Id = int(tile.id)
        self.Shape = Shape(tile.shape)
        self.Color = Color(tile.color)
        self.RackPosition = tile.rackPosition

File: Qwirkle.Domain/ValueObjects/TileOnPlayer.cs
namespace Qwirkle.Domain.ValueObjects;

public record TileOnPlayer(RackPosition RackPosition, TileColor Color, TileShape Shape) : Tile(Color, Shape)
{
    public TileOnBoard ToTileOnBoard(Coordinates coordinates) => TileOnBoard.From(this, coordinates);
    public Tile ToTile() => new(Color, Shape);
}
